src/query/service/src/interpreters/interpreter_table_create.rs

```diff
@@ -405,6 +406,21 @@ pub fn is_valid_block_per_segment(options: &BTreeMap<String, String>) -> Result<
             return Err(ErrorCode::TableOptionInvalid(error_str));
         }
     }
+
+    Ok(())
+}
+
+pub fn is_valid_row_per_block(options: &BTreeMap<String, String>) -> Result<()> {
+    // check row_per_block can not be over 1000000.
+    if let Some(value) = options.get(FUSE_OPT_KEY_ROW_PER_BLOCK) {
+        let row_per_block = value.parse::<u64>()?;
+        let error_str = "invalid row_per_block option, can't be over 1000000";
+
+        if row_per_block > DEFAULT_BLOCK_MAX_ROWS as u64 {
```

Comment:
BTW check the `DEFAULT_BLOCK_MIN_ROWS` of blocks?

Reply:
> DEFAULT_BLOCK_MIN_ROWS

The `block_min_rows` does not need to be checked because its value is  `row_per_block`  * 0.8.